build-coordinator/src/main/java/org/jboss/pnc/core/builder/coordinator/BuildScheduler.java

```diff
@@ -33,7 +33,7 @@
  * @author <a href="mailto:matejonnet@gmail.com">Matej Lazar</a>
  */
 public interface BuildScheduler {
-    void startBuilding(BuildTask buildTask, Consumer<BuildStatus> onComplete) throws CoreException;
+    void startBuilding(BuildTask buildTask, Consumer<BuildResult> onComplete) throws CoreException, ExecutorException;
```

Comment:
From client code perspective - why do we need 2 exceptions here? Both are unrecoverable...